src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.cpp

```diff
@@ -0,0 +1,338 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.hpp"
+
+#include <cmath>
+#include <pup.h>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/TempBuffer.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/Variables.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace {
+// compute gauge wave H
+template <typename DataType>
+DataType gauge_wave_h(const tnsr::I<DataType, 3>& x, const double t,
+                      const double amplitude,
+                      const double wavelength) noexcept {
+  return {-1.0 * amplitude * sin((get<0>(x) - t) * (2.0 * M_PI / wavelength)) +
+          1.0};
+}
+
+// compute gauge wave derivH: \partial_x H
+template <typename DataType>
+DataType gauge_wave_deriv_h(const tnsr::I<DataType, 3>& x, const double t,
+                            const double amplitude,
+                            const double wavelength) noexcept {
+  return {-1.0 * amplitude * (2.0 * M_PI / wavelength) *
+          cos((get<0>(x) - t) * (2.0 * M_PI / wavelength))};
+}
+}  // namespace
+
+namespace gr {
+namespace Solutions {
+GaugeWave::GaugeWave() noexcept {
+  amplitude_ = 1.0;
+  wavelength_ = 1.0;
+}
+
+GaugeWave::GaugeWave(const double amplitude, const double wavelength) noexcept
+    : amplitude_(amplitude), wavelength_(wavelength) {
+  ASSERT(amplitude > 0.0,
+         "Amplitude must be non-negative. Given amplitude: " << amplitude_);
+  ASSERT(wavelength > 0.0,
+         "Wavelength must be non-negative. Given wavelength: " << wavelength_);
+}
+
+void GaugeWave::pup(PUP::er& p) noexcept {
+  p | amplitude_;
+  p | wavelength_;
+}
+
+template <typename DataType>
+GaugeWave::IntermediateVars<DataType>::IntermediateVars(
+    const double amplitude, const double wavelength,
+    const tnsr::I<DataType, GaugeWave::volume_dim, Frame::Inertial>& x,
+    double t) noexcept {
+  h = gauge_wave_h(x, t, amplitude, wavelength);
+  dx_h = gauge_wave_deriv_h(x, t, amplitude, wavelength);
+  sqrt_h = sqrt(h);
+  dx_h_over_2_sqrt_h = 0.5 * dx_h / sqrt(h);
+}
+
+template <typename DataType>
+auto GaugeWave::variables(
+    const tnsr::I<DataType, GaugeWave::volume_dim, Frame::Inertial>& /*x*/,
+    double /*t*/, const GaugeWave::IntermediateVars<DataType>& vars,
+    tmpl::list<gr::Tags::Lapse<DataType>> /*meta*/) const noexcept
+    -> tuples::TaggedTuple<gr::Tags::Lapse<DataType>> {
+  return {Scalar<DataType>{DataType{vars.sqrt_h}}};
```

Comment:
Seems to work...I just saw this done in BondiMichel, and I assumed that since that code was reviewed within the past year, that there was some reason for it.